src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -2025,31 +2025,45 @@ def get_zones(self):
 
     def print_zone(self, zone):
         """ Prints all the contents of a zone file, takes domain.com or 1.168.192.in-addr.arpa string """
-        zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone} | grep -v \; | sed "s/\t/ /g" | grep e |'
-                                        f'sed "s/\   / /g"')
+        zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone}')
+#        zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone} | grep -v \; | sed "s/\t/ /g" | grep e |'
+#                                        f'sed "s/\   / /g"')
         return zone.stdout_text
 
     def find_a(self, hostname):
         """ Searches the zone for a specific A record, returning True for success and False for failure """
         domain = self.ad_host.realm.lower()
         host = hostname.split(".")[0]
-        results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | grep e |'
-                                           f'sed "s/\   / /g" | grep {host}', raiseonerr=False)
-        if host in results.stdout_text:
-            return True
-        return False
+        seconds = 0
+        while seconds <= 300:
+            results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | '
```

Comment:
Please remove all of the grep and sed because:
1. It is not needed on the AD host for some next command.
2. It will complain about deprecated expressions.
3. You are introducing dependency on cygwing for these tests.
4. You can do all of the string changes in python instead.

Reply:
```
2023-01-05T18:30:07 =============================== warnings summary ===============================
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2028
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2028: DeprecationWarning: invalid escape sequence '\;'
2023-01-05T18:30:07     zone = self.ad_host.run_command(f'dnscmd.exe /zoneprint {zone} | grep -v \; | sed "s/\t/ /g" | grep e |'
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2029
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2029: DeprecationWarning: invalid escape sequence '\ '
2023-01-05T18:30:07     f'sed "s/\   / /g"')
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2036
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2036: DeprecationWarning: invalid escape sequence '\;'
2023-01-05T18:30:07     results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | grep e |'
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2037
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2037: DeprecationWarning: invalid escape sequence '\ '
2023-01-05T18:30:07     f'sed "s/\   / /g" | grep {host}', raiseonerr=False)
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2047
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2047: DeprecationWarning: invalid escape sequence '\;'
2023-01-05T18:30:07     results = self.ad_host.run_command(f'dnscmd.exe /zoneprint {net} | grep -v \; | sed "s/\t/ /g" | grep e |'
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/sssd/testlib/common/utils.py:2048
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/sssd/testlib/common/utils.py:2048: DeprecationWarning: invalid escape sequence '\ '
2023-01-05T18:30:07     f'sed "s/\   / /g" | grep -v hostmaster | grep {hostname}',
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/ad/test_dyndns.py:167
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/ad/test_dyndns.py:167: DeprecationWarning: invalid escape sequence '\;'
2023-01-05T18:30:07     cmd = f'dnscmd.exe /zoneprint {domain} | grep -v \; | sed "s/\t/ /g" | grep e | ' \
2023-01-05T18:30:07 
2023-01-05T18:30:07 src/tests/multihost/ad/test_dyndns.py:168
2023-01-05T18:30:07   /home/jenkins/workspace/User-Tools/trigger-test-suite-tool/sssd/src/tests/multihost/ad/test_dyndns.py:168: DeprecationWarning: invalid escape sequence '\ '
2023-01-05T18:30:07     f'sed "s/\   / /g" | grep {hostname.split(".")[0]}'
```

Reply:
Ack, are you okay with me doing this later? For right now, I want to get the tests working and green.

Reply:
Please at least re-evaluate which one are really needed. In most cases You are just checking that hostname is or is not present right? My guess is that lines with ";" are comments (and should not contain hostname), all of the replacing tabs does not look like it will make any difference and grep {hostname.split(".")[0]}  stuff can be if x in y:.

Reply:
'grep -v ";"' should work the same and python might not complain, but I would rather not have cygwin as a dependency as we might want, at some point, switch to windows native ssh server.

Reply:
I agree, we need to remove this dependency. I will re-evalulate it and provide changes.

Reply:
I removed it, the tests are passing. 

https://ci-jenkins-csb-idmops.apps.ocp-c1.prod.psi.redhat.com/job/User-Tools/job/trigger-test-suite-tool/803/console